index.js

```diff
@@ -521,13 +557,14 @@ var Agent = {
         if (cachedAgent) {
             return cachedAgent[VERSION];
         }
+        let internalUsername = cachedUsernames[username];
```

Comment:
We need to move this to above line 556 and then do the lookup in cachedAgents against internalUsername rather than username.